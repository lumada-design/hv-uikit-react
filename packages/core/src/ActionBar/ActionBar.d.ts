import * as React from "react";
import { StandardProps } from "@mui/material";

export type HvActionContainerClassKey = "root";

export type HvActionContainerProps = StandardProps<
  React.HTMLAttributes<HTMLDivElement>,
  HvActionContainerClassKey
>;

export default function HvActionContainer(props: HvActionContainerProps): JSX.Element | null;
