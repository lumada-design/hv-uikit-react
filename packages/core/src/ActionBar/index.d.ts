export { default } from "./ActionBar";
export * from "./ActionBar";
