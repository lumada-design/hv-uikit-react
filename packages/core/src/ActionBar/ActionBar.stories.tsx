import { css } from "@emotion/css";
import { DecoratorFn, Meta, StoryObj } from "@storybook/react";
import {
  HvActionBar,
  HvActionBarProps,
  HvButton,
  theme,
} from "@hitachivantara/uikit-react-core";

const containerDecorator: DecoratorFn = (Story) => (
  <div
    className={css({
      position: "relative",
      margin: "auto",
      backgroundColor: theme.colors.atmo1,
      height: "150px",
      width: "400px",
      display: "flex",
      alignItems: "flex-end",
    })}
  >
    {Story()}
  </div>
);

const meta: Meta<typeof HvActionBar> = {
  title: "Widgets/Action Bar",
  component: HvActionBar,
  decorators: [containerDecorator],
};

export default meta;

export const Main: StoryObj<HvActionBarProps> = {
  argTypes: {
    classes: { control: { disable: true } },
  },
  render: () => {
    return (
      <HvActionBar className={css({ gap: theme.space.xs })}>
        <HvButton variant="secondaryGhost">Help</HvButton>
        <div style={{ flex: 1 }} aria-hidden="true">
          &nbsp;
        </div>
        <HvButton variant="secondaryGhost">Save</HvButton>
        <HvButton variant="secondaryGhost">Cancel</HvButton>
      </HvActionBar>
    );
  },
};

export const DualAction: StoryObj<HvActionBarProps> = {
  parameters: {
    docs: {
      description: {
        story: "Showcasing the action bar pattern with only two actions.",
      },
    },
  },
  render: () => {
    return (
      <HvActionBar className={css({ gap: theme.space.xs })}>
        <HvButton variant="secondaryGhost">Save</HvButton>
        <HvButton variant="secondaryGhost">Cancel</HvButton>
      </HvActionBar>
    );
  },
};
