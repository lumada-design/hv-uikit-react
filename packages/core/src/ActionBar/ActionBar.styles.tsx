import { theme } from "@hitachivantara/uikit-styles";

import { createClasses } from "../utils/classes";

export const { staticClasses, useClasses } = createClasses("HvActionBar", {
  root: {
    width: "100%",
    padding: theme.space.sm,
    borderTop: `1px solid ${theme.colors.atmo4}`,
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
  },
});
