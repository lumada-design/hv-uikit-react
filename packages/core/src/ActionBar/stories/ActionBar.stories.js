import React from "react";
import { makeStyles } from "@material-ui/core";
import { HvActionContainer, HvButton, HvDropDownMenu } from "../..";

export default {
  title: "Patterns/Action Bar/Action Container",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvActionContainer } from '@hv/uikit-react-core/dist'"
  },
  component: HvActionContainer
};

export const Main = () => {
  const useStyles = makeStyles(theme => ({
    container: {
      position: "relative",
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      height: "150px",
      width: "400px"
    },
    actionBar: {
      position: "absolute",
      bottom: "0"
    },
    space: {
      flex: 1
    },
    buttonSeparator: {
      marginRight: theme.hv.spacing.xs
    }
  }));

  const classes = useStyles();
  return (
    <div className={classes.container}>
      <HvActionContainer className={classes.actionBar}>
        <HvButton category="ghost" onClick={() => console.log("Help action")}>
          Help
        </HvButton>
        <div aria-hidden="true" className={classes.space}>
          &nbsp;
        </div>
        <HvButton
          className={classes.buttonSeparator}
          category="ghost"
          onClick={() => console.log("Save action")}
        >
          Save
        </HvButton>
        <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
          Cancel
        </HvButton>
      </HvActionContainer>
    </div>
  );
};

export const DualAction = () => {
  const useStyles = makeStyles(theme => ({
    container: {
      position: "relative",
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      height: "150px",
      width: "400px"
    },
    actionBar: {
      position: "absolute",
      bottom: "0"
    },
    space: {
      flex: 1
    },
    buttonSeparator: {
      marginRight: theme.hv.spacing.xs
    }
  }));

  const classes = useStyles();
  return (
    <div className={classes.container}>
      <HvActionContainer className={classes.actionBar}>
        <HvButton
          className={classes.buttonSeparator}
          category="ghost"
          onClick={() => console.log("Save action")}
        >
          Save
        </HvButton>
        <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
          Cancel
        </HvButton>
      </HvActionContainer>
    </div>
  );
};

DualAction.parameters = {
  docs: {
    description: { story: "Showcasing the action bar pattern with only two actions." }
  }
};

export const VariedActionBar = () => {
  const useStyles = makeStyles(theme => ({
    container: {
      position: "relative",
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      height: "150px",
      width: "400px"
    },
    actionBar: {
      position: "absolute",
      bottom: "0"
    },
    space: {
      flex: 1
    },
    buttonSeparator: {
      marginRight: theme.hv.spacing.xs
    }
  }));

  const classes = useStyles();
  return (
    <div className={classes.container}>
      <HvActionContainer className={classes.actionBar}>
        <HvButton category="ghost" onClick={() => console.log("Help action")}>
          Help
        </HvButton>
        <div aria-hidden="true" className={classes.space}>
          &nbsp;
        </div>
        <HvButton
          className={classes.buttonSeparator}
          category="ghost"
          onClick={() => console.log("Save action")}
        >
          Save
        </HvButton>
        <HvDropDownMenu
          onClick={(e, item) => console.log(item.label)}
          dataList={[{ label: "Delete" }, { label: "Update" }]}
        />
      </HvActionContainer>
    </div>
  );
};

VariedActionBar.parameters = {
  docs: {
    description: { story: "Showcasing the action bar pattern with only two actions." }
  }
};
