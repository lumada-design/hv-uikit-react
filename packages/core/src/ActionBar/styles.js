const styles = (theme) => ({
  root: {
    width: "100%",
    height: "75px",
    padding: theme.hv.spacing.sm,
    borderTop: `3px solid ${theme.hv.palette.atmosphere.atmo2}`,
    display: "flex",
    justifyContent: "flex-end",
  },
});

export default styles;
