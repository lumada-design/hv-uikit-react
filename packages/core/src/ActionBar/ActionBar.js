import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { withStyles } from "@mui/styles";
import styles from "./styles";

/**
 * This component acts as a container for the Action Bar pattern.
 */
const HvActionBar = (props) => {
  const { classes, className, id, children, ...others } = props;

  return (
    <div id={id} className={clsx(classes.root, className)} {...others}>
      {children}
    </div>
  );
};

HvActionBar.propTypes = {
  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the component root class.
     */
    root: PropTypes.string,
  }).isRequired,
  /**
   * The content of the action container.
   */
  children: PropTypes.node.isRequired,
};

export default withStyles(styles, { name: "HvActionBar" })(HvActionBar);
