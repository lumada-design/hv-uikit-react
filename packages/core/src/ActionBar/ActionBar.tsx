import { HvBaseProps } from "../types/generic";
import { useDefaultProps } from "../hooks/useDefaultProps";

import { ExtractNames } from "../utils/classes";

import { staticClasses, useClasses } from "./ActionBar.styles";

export { staticClasses as actionBarClasses };

export type HvActionBarClasses = ExtractNames<typeof useClasses>;

export interface HvActionBarProps extends HvBaseProps {
  /** A Jss Object used to override or extend the styles applied to the component. */
  classes?: HvActionBarClasses;
}

export const HvActionBar = (props: HvActionBarProps) => {
  const {
    classes: classesProp,
    className,
    children,
    ...others
  } = useDefaultProps("HvActionBar", props);
  const { classes, cx } = useClasses(classesProp);

  return (
    <div className={cx(classes.root, className)} {...others}>
      {children}
    </div>
  );
};
