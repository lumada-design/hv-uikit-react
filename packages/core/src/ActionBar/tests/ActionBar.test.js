import React from "react";
import { mount } from "enzyme";
import { Main } from "../stories/ActionBar.stories";
import { HvActionBar, HvProvider } from "../..";

describe("Action Bar", () => {
  const wrapper = mount(
    <HvProvider>
      <Main />
    </HvProvider>
  );

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(HvActionBar)).toMatchSnapshot();
  });
});
