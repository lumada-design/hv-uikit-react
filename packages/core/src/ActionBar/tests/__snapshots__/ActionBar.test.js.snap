// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Bar should render correctly 1`] = `
<WithStyles(HvActionBar)
  className="makeStyles-actionBar-2"
>
  <HvActionBar
    className="makeStyles-actionBar-2"
    classes={
      Object {
        "root": "HvActionBar-root",
      }
    }
  >
    <div
      className="HvActionBar-root makeStyles-actionBar-2"
    >
      <WithStyles(HvButton)
        category="ghost"
        onClick={[Function]}
      >
        <HvButton
          category="ghost"
          classes={
            Object {
              "focusVisible": "HvButton-focusVisible",
              "ghost": "HvButton-ghost",
              "ghostDisabled": "HvButton-ghostDisabled",
              "ghostSVG": "HvButton-ghostSVG",
              "icon": "HvButton-icon",
              "iconSVG": "HvButton-iconSVG",
              "primary": "HvButton-primary",
              "primaryDisabled": "HvButton-primaryDisabled",
              "primarySVG": "HvButton-primarySVG",
              "root": "HvButton-root",
              "secondary": "HvButton-secondary",
              "secondaryDisabled": "HvButton-secondaryDisabled",
              "secondarySVG": "HvButton-secondarySVG",
              "semantic": "HvButton-semantic",
              "semanticDisabled": "HvButton-semanticDisabled",
              "semanticSVG": "HvButton-semanticSVG",
              "startIcon": "HvButton-startIcon",
            }
          }
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Button))
            className="HvButton-ghostSVG"
            classes={
              Object {
                "containedPrimary": "HvButton-primary",
                "disabled": "HvButton-ghostDisabled",
                "outlinedPrimary": "HvButton-secondary",
                "root": "HvButton-root",
                "startIcon": "HvButton-startIcon",
                "textPrimary": "HvButton-ghost",
              }
            }
            color="primary"
            disableRipple={true}
            focusVisibleClassName="HvButton-focusVisible"
            onClick={[Function]}
            variant="text"
          >
            <ForwardRef(Button)
              className="HvButton-ghostSVG"
              classes={
                Object {
                  "colorInherit": "MuiButton-colorInherit",
                  "contained": "MuiButton-contained",
                  "containedPrimary": "MuiButton-containedPrimary HvButton-primary",
                  "containedSecondary": "MuiButton-containedSecondary",
                  "containedSizeLarge": "MuiButton-containedSizeLarge",
                  "containedSizeSmall": "MuiButton-containedSizeSmall",
                  "disableElevation": "MuiButton-disableElevation",
                  "disabled": "Mui-disabled HvButton-ghostDisabled",
                  "endIcon": "MuiButton-endIcon",
                  "focusVisible": "Mui-focusVisible",
                  "fullWidth": "MuiButton-fullWidth",
                  "iconSizeLarge": "MuiButton-iconSizeLarge",
                  "iconSizeMedium": "MuiButton-iconSizeMedium",
                  "iconSizeSmall": "MuiButton-iconSizeSmall",
                  "label": "MuiButton-label",
                  "outlined": "MuiButton-outlined",
                  "outlinedPrimary": "MuiButton-outlinedPrimary HvButton-secondary",
                  "outlinedSecondary": "MuiButton-outlinedSecondary",
                  "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                  "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                  "root": "MuiButton-root HvButton-root",
                  "sizeLarge": "MuiButton-sizeLarge",
                  "sizeSmall": "MuiButton-sizeSmall",
                  "startIcon": "MuiButton-startIcon HvButton-startIcon",
                  "text": "MuiButton-text",
                  "textPrimary": "MuiButton-textPrimary HvButton-ghost",
                  "textSecondary": "MuiButton-textSecondary",
                  "textSizeLarge": "MuiButton-textSizeLarge",
                  "textSizeSmall": "MuiButton-textSizeSmall",
                }
              }
              color="primary"
              disableRipple={true}
              focusVisibleClassName="HvButton-focusVisible"
              onClick={[Function]}
              variant="text"
            >
              <WithStyles(ForwardRef(ButtonBase))
                className="MuiButton-root HvButton-root MuiButton-text HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                component="button"
                disableRipple={true}
                disabled={false}
                focusRipple={true}
                focusVisibleClassName="Mui-focusVisible HvButton-focusVisible"
                onClick={[Function]}
                type="button"
              >
                <ForwardRef(ButtonBase)
                  className="MuiButton-root HvButton-root MuiButton-text HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "root": "MuiButtonBase-root",
                    }
                  }
                  component="button"
                  disableRipple={true}
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible HvButton-focusVisible"
                  onClick={[Function]}
                  type="button"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Help
                    </span>
                  </button>
                </ForwardRef(ButtonBase)>
              </WithStyles(ForwardRef(ButtonBase))>
            </ForwardRef(Button)>
          </WithStyles(ForwardRef(Button))>
        </HvButton>
      </WithStyles(HvButton)>
      <div
        aria-hidden="true"
        className="makeStyles-space-3"
      >
        Â 
      </div>
      <WithStyles(HvButton)
        category="ghost"
        className="makeStyles-buttonSeparator-4"
        onClick={[Function]}
      >
        <HvButton
          category="ghost"
          className="makeStyles-buttonSeparator-4"
          classes={
            Object {
              "focusVisible": "HvButton-focusVisible",
              "ghost": "HvButton-ghost",
              "ghostDisabled": "HvButton-ghostDisabled",
              "ghostSVG": "HvButton-ghostSVG",
              "icon": "HvButton-icon",
              "iconSVG": "HvButton-iconSVG",
              "primary": "HvButton-primary",
              "primaryDisabled": "HvButton-primaryDisabled",
              "primarySVG": "HvButton-primarySVG",
              "root": "HvButton-root",
              "secondary": "HvButton-secondary",
              "secondaryDisabled": "HvButton-secondaryDisabled",
              "secondarySVG": "HvButton-secondarySVG",
              "semantic": "HvButton-semantic",
              "semanticDisabled": "HvButton-semanticDisabled",
              "semanticSVG": "HvButton-semanticSVG",
              "startIcon": "HvButton-startIcon",
            }
          }
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Button))
            className="makeStyles-buttonSeparator-4 HvButton-ghostSVG"
            classes={
              Object {
                "containedPrimary": "HvButton-primary",
                "disabled": "HvButton-ghostDisabled",
                "outlinedPrimary": "HvButton-secondary",
                "root": "HvButton-root",
                "startIcon": "HvButton-startIcon",
                "textPrimary": "HvButton-ghost",
              }
            }
            color="primary"
            disableRipple={true}
            focusVisibleClassName="HvButton-focusVisible"
            onClick={[Function]}
            variant="text"
          >
            <ForwardRef(Button)
              className="makeStyles-buttonSeparator-4 HvButton-ghostSVG"
              classes={
                Object {
                  "colorInherit": "MuiButton-colorInherit",
                  "contained": "MuiButton-contained",
                  "containedPrimary": "MuiButton-containedPrimary HvButton-primary",
                  "containedSecondary": "MuiButton-containedSecondary",
                  "containedSizeLarge": "MuiButton-containedSizeLarge",
                  "containedSizeSmall": "MuiButton-containedSizeSmall",
                  "disableElevation": "MuiButton-disableElevation",
                  "disabled": "Mui-disabled HvButton-ghostDisabled",
                  "endIcon": "MuiButton-endIcon",
                  "focusVisible": "Mui-focusVisible",
                  "fullWidth": "MuiButton-fullWidth",
                  "iconSizeLarge": "MuiButton-iconSizeLarge",
                  "iconSizeMedium": "MuiButton-iconSizeMedium",
                  "iconSizeSmall": "MuiButton-iconSizeSmall",
                  "label": "MuiButton-label",
                  "outlined": "MuiButton-outlined",
                  "outlinedPrimary": "MuiButton-outlinedPrimary HvButton-secondary",
                  "outlinedSecondary": "MuiButton-outlinedSecondary",
                  "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                  "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                  "root": "MuiButton-root HvButton-root",
                  "sizeLarge": "MuiButton-sizeLarge",
                  "sizeSmall": "MuiButton-sizeSmall",
                  "startIcon": "MuiButton-startIcon HvButton-startIcon",
                  "text": "MuiButton-text",
                  "textPrimary": "MuiButton-textPrimary HvButton-ghost",
                  "textSecondary": "MuiButton-textSecondary",
                  "textSizeLarge": "MuiButton-textSizeLarge",
                  "textSizeSmall": "MuiButton-textSizeSmall",
                }
              }
              color="primary"
              disableRipple={true}
              focusVisibleClassName="HvButton-focusVisible"
              onClick={[Function]}
              variant="text"
            >
              <WithStyles(ForwardRef(ButtonBase))
                className="MuiButton-root HvButton-root MuiButton-text makeStyles-buttonSeparator-4 HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                component="button"
                disableRipple={true}
                disabled={false}
                focusRipple={true}
                focusVisibleClassName="Mui-focusVisible HvButton-focusVisible"
                onClick={[Function]}
                type="button"
              >
                <ForwardRef(ButtonBase)
                  className="MuiButton-root HvButton-root MuiButton-text makeStyles-buttonSeparator-4 HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "root": "MuiButtonBase-root",
                    }
                  }
                  component="button"
                  disableRipple={true}
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible HvButton-focusVisible"
                  onClick={[Function]}
                  type="button"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text makeStyles-buttonSeparator-4 HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Save
                    </span>
                  </button>
                </ForwardRef(ButtonBase)>
              </WithStyles(ForwardRef(ButtonBase))>
            </ForwardRef(Button)>
          </WithStyles(ForwardRef(Button))>
        </HvButton>
      </WithStyles(HvButton)>
      <WithStyles(HvButton)
        category="ghost"
        onClick={[Function]}
      >
        <HvButton
          category="ghost"
          classes={
            Object {
              "focusVisible": "HvButton-focusVisible",
              "ghost": "HvButton-ghost",
              "ghostDisabled": "HvButton-ghostDisabled",
              "ghostSVG": "HvButton-ghostSVG",
              "icon": "HvButton-icon",
              "iconSVG": "HvButton-iconSVG",
              "primary": "HvButton-primary",
              "primaryDisabled": "HvButton-primaryDisabled",
              "primarySVG": "HvButton-primarySVG",
              "root": "HvButton-root",
              "secondary": "HvButton-secondary",
              "secondaryDisabled": "HvButton-secondaryDisabled",
              "secondarySVG": "HvButton-secondarySVG",
              "semantic": "HvButton-semantic",
              "semanticDisabled": "HvButton-semanticDisabled",
              "semanticSVG": "HvButton-semanticSVG",
              "startIcon": "HvButton-startIcon",
            }
          }
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Button))
            className="HvButton-ghostSVG"
            classes={
              Object {
                "containedPrimary": "HvButton-primary",
                "disabled": "HvButton-ghostDisabled",
                "outlinedPrimary": "HvButton-secondary",
                "root": "HvButton-root",
                "startIcon": "HvButton-startIcon",
                "textPrimary": "HvButton-ghost",
              }
            }
            color="primary"
            disableRipple={true}
            focusVisibleClassName="HvButton-focusVisible"
            onClick={[Function]}
            variant="text"
          >
            <ForwardRef(Button)
              className="HvButton-ghostSVG"
              classes={
                Object {
                  "colorInherit": "MuiButton-colorInherit",
                  "contained": "MuiButton-contained",
                  "containedPrimary": "MuiButton-containedPrimary HvButton-primary",
                  "containedSecondary": "MuiButton-containedSecondary",
                  "containedSizeLarge": "MuiButton-containedSizeLarge",
                  "containedSizeSmall": "MuiButton-containedSizeSmall",
                  "disableElevation": "MuiButton-disableElevation",
                  "disabled": "Mui-disabled HvButton-ghostDisabled",
                  "endIcon": "MuiButton-endIcon",
                  "focusVisible": "Mui-focusVisible",
                  "fullWidth": "MuiButton-fullWidth",
                  "iconSizeLarge": "MuiButton-iconSizeLarge",
                  "iconSizeMedium": "MuiButton-iconSizeMedium",
                  "iconSizeSmall": "MuiButton-iconSizeSmall",
                  "label": "MuiButton-label",
                  "outlined": "MuiButton-outlined",
                  "outlinedPrimary": "MuiButton-outlinedPrimary HvButton-secondary",
                  "outlinedSecondary": "MuiButton-outlinedSecondary",
                  "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                  "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                  "root": "MuiButton-root HvButton-root",
                  "sizeLarge": "MuiButton-sizeLarge",
                  "sizeSmall": "MuiButton-sizeSmall",
                  "startIcon": "MuiButton-startIcon HvButton-startIcon",
                  "text": "MuiButton-text",
                  "textPrimary": "MuiButton-textPrimary HvButton-ghost",
                  "textSecondary": "MuiButton-textSecondary",
                  "textSizeLarge": "MuiButton-textSizeLarge",
                  "textSizeSmall": "MuiButton-textSizeSmall",
                }
              }
              color="primary"
              disableRipple={true}
              focusVisibleClassName="HvButton-focusVisible"
              onClick={[Function]}
              variant="text"
            >
              <WithStyles(ForwardRef(ButtonBase))
                className="MuiButton-root HvButton-root MuiButton-text HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                component="button"
                disableRipple={true}
                disabled={false}
                focusRipple={true}
                focusVisibleClassName="Mui-focusVisible HvButton-focusVisible"
                onClick={[Function]}
                type="button"
              >
                <ForwardRef(ButtonBase)
                  className="MuiButton-root HvButton-root MuiButton-text HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "root": "MuiButtonBase-root",
                    }
                  }
                  component="button"
                  disableRipple={true}
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible HvButton-focusVisible"
                  onClick={[Function]}
                  type="button"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Cancel
                    </span>
                  </button>
                </ForwardRef(ButtonBase)>
              </WithStyles(ForwardRef(ButtonBase))>
            </ForwardRef(Button)>
          </WithStyles(ForwardRef(Button))>
        </HvButton>
      </WithStyles(HvButton)>
    </div>
  </HvActionBar>
</WithStyles(HvActionBar)>
`;
